GetBPDCitizen/handler.ts

```diff
@@ -0,0 +1,143 @@
+import * as express from "express";
+
+import { Context } from "@azure/functions";
+import { isEmpty } from "fp-ts/lib/Array";
+import { Either, isLeft } from "fp-ts/lib/Either";
+import { isNone, Option } from "fp-ts/lib/Option";
+import { TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import * as t from "io-ts";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseErrorValidation,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { Repository } from "typeorm";
+import { BPDCitizen } from "../generated/definitions/BPDCitizen";
+import { Citizen } from "../models/citizen";
+import { OptionalHeaderMiddleware } from "../utils/middleware/optional_header";
+
+type IHttpHandler = (
+  context: Context,
+  requestFiscalCode: Option<FiscalCode>
+) => Promise<
+  // tslint:disable-next-line: max-union-size
+  | IResponseSuccessJson<BPDCitizen>
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+  | IResponseErrorValidation
+>;
+
+// Convert model object to API object
+export const toApiBPDCitizen = (
+  domainObj: ReadonlyArray<Citizen>
+): Either<t.Errors, BPDCitizen> => {
+  return BPDCitizen.decode(
+    domainObj.reduce((acc: BPDCitizen | undefined, citizen) => {
+      if (acc === undefined) {
+        return {
+          citizen_enabled: citizen.citizen_enabled,
+          fiscal_code: citizen.fiscal_code,
+          onboarding_date: citizen.onboarding_date?.toISOString(),
+          onboarding_issuer_id: citizen.onboarding_issuer_id,
+          payment_methods:
+            citizen.payment_instrument_hpan && citizen.payment_instrument_status
+              ? [
+                  {
+                    hpan: citizen.payment_instrument_hpan,
+                    status: citizen.payment_instrument_status
+                  }
```

Comment:
why this name mapping? can't we just return `payment_instrument_hpan` and `payment_instrument_status`?
(the principle is to keep the backend as simple as it can)

Reply:
Fixed in 05a99a9